src/lib/homestore/replication_state_machine.cpp

```diff
@@ -23,36 +28,51 @@ void ReplicationStateMachine::on_commit(int64_t lsn, const sisl::blob& header, c
 
 void ReplicationStateMachine::on_pre_commit(int64_t lsn, sisl::blob const& header, sisl::blob const& key, void* ctx) {
     LOGINFO("on_pre_commit with lsn:{}", lsn);
-    const ReplicationMessageHeader* msg_header = r_cast< const ReplicationMessageHeader* >(header.bytes);
-
-    switch (msg_header->message_type) {
-    case ReplicationMessageType::SHARD_MESSAGE: {
-        _home_object->on_pre_commit_shard_msg(lsn, header, key, ctx);
-        break;
-    }
-    case ReplicationMessageType::PG_MESSAGE:
-    case ReplicationMessageType::BLOB_MESSAGE:
-    default: {
-        break;
-    }
-    }
+    // For shard creation, since homestore repldev inside will write shard header to data service first before this
```

Comment:
I think we still need to track the _flying_shards and checks the uniqueness of shard.
The shard ID is given by master,  if leader change the new leader might propose a shardID that had been proposed.